commit 22b738ac3309c8712ac6aa8095e27bbc4067e4b3
Author: James Smart <jsmart2021@gmail.com>
Date:   Tue Mar 12 16:30:11 2019 -0700

    scsi: lpfc: Fix nvmet handling of first burst cmd
    
    With negative test injection, the driver is receiving a command with first
    burst enabled, meaning Sequence initiative is not passed with the command
    frame. The driver notes the condition and discards the frame.  However the
    driver calls the incorrect buffer free routine, resulting in a NULL pointer
    reference.
    
    For hbq buffer free, convert to using lpfc_rq_buf_free().
    
    Signed-off-by: Dick Kennedy <dick.kennedy@broadcom.com>
    Signed-off-by: James Smart <jsmart2021@gmail.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/lpfc/lpfc_sli.c b/drivers/scsi/lpfc/lpfc_sli.c
index de5715b07522..c82304c11ef2 100644
--- a/drivers/scsi/lpfc/lpfc_sli.c
+++ b/drivers/scsi/lpfc/lpfc_sli.c
@@ -14019,7 +14019,7 @@ lpfc_sli4_nvmet_handle_rcqe(struct lpfc_hba *phba, struct lpfc_queue *cq,
 			return false;
 		}
 drop:
-		lpfc_in_buf_free(phba, &dma_buf->dbuf);
+		lpfc_rq_buf_free(phba, &dma_buf->hbuf);
 		break;
 	case FC_STATUS_INSUFF_BUF_FRM_DISC:
 		if (phba->nvmet_support) {